src/main/java/com/synopsys/integration/detect/lifecycle/run/RunManager.java

```diff
@@ -141,6 +156,8 @@ public RunResult run(ProductRunData productRunData) throws DetectUserFriendlyExc
         BdioCodeLocationCreator bdioCodeLocationCreator = detectContext.getBean(BdioCodeLocationCreator.class);
         DetectInfo detectInfo = detectContext.getBean(DetectInfo.class);
         DetectDetectableFactory detectDetectableFactory = detectContext.getBean(DetectDetectableFactory.class);
+        blackDuckServicesFactory = productRunData.getBlackDuckRunData().getBlackDuckServicesFactory().orElse(null); // TODO - what is best way to provide blackduck service to populateEnabledTools?
+        enabledTools = fetchEnabledTools();
```

Comment:
Fetching enabled tools should be done in Boot. Likely in blackduck connectivity checker, where we check for permissions. Also I think this is a job for DetectToolFilter, you could subclass it for FeatureEnabledDetectToolFilter and only when online and connected to Black Duck does the subclass get used.